crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -33,154 +33,95 @@ pub struct CheckpointObjectChanges {
 }
 
 #[async_trait]
-impl Worker for ObjectsSnapshotProcessor {
+impl Worker for ObjectsSnapshotHandler {
     async fn process_checkpoint(&self, checkpoint: &CheckpointData) -> anyhow::Result<()> {
-        let checkpoint_sequence_number = checkpoint.checkpoint_summary.sequence_number;
-        // Index the object changes and send them to the committer.
-        let object_changes: TransactionObjectChangesToCommit =
-            CheckpointHandler::index_objects(checkpoint, &self.metrics).await?;
-        self.indexed_obj_sender
-            .send(CheckpointObjectChanges {
-                checkpoint_sequence_number,
-                object_changes,
-            })
+        let transformed_data = CheckpointHandler::index_objects(checkpoint, &self.metrics).await?;
+        self.sender
+            .send((
+                checkpoint.checkpoint_summary.sequence_number,
+                transformed_data,
+            ))
             .await?;
         Ok(())
     }
 }
 
-// Start both the ingestion pipeline and committer for objects snapshot table.
-pub async fn start_objects_snapshot_processor(
+#[async_trait]
+impl Handler<TransactionObjectChangesToCommit> for ObjectsSnapshotHandler {
+    fn name(&self) -> String {
+        "objects_snapshot_handler".to_string()
+    }
+
+    async fn load(
+        &self,
+        transformed_data: Vec<TransactionObjectChangesToCommit>,
+    ) -> IndexerResult<()> {
+        self.store
+            .persist_objects_snapshot(transformed_data)
```

Comment:
This PR is in a good shape to merge. But I think the work is not fully done here for this pipeline.
The processing only transforms the checkpoint data to a intermediate data type, which then gets transformed again within `persist_objects_snapshot`.
This creates a few inefficiencies:
1. More processing could have been done in parallel.
2. It's difficult to tune `checkpoint_commit_batch_size` because the actual commit size depends on the amount of data within the intermediate data type.
3. There are still a lot of boiler-plate code inside those persist_ functions

So ideally we should have the processing to return `Vec<StoredObjectSnapshot>` directly. Then we could apply proper chunk within the framework, so that the load function only need to worry about the diesel/sql schema.